jdaviz/configs/cubeviz/plugins/viewers.py

```diff
@@ -78,14 +79,19 @@ def slice_values(self):
         take_inds.remove(self.slice_index)
         for layer in self.layers:
             try:
-                data_obj = layer.layer.data.get_component(self.slice_component_label).data
+                spec_axis = layer.layer.data.get_object(cls=Spectrum1D).spectral_axis
+                display_spectral_units = self.jdaviz_app._get_display_unit('spectral')
+                if display_spectral_units is not '' and spec_axis.unit != display_spectral_units:
+                    converted_axis = spec_axis.to(display_spectral_units).value
+                else:
+                    converted_axis = spec_axis.value
```

Comment:
This is making slice selection much slower but I do not have a great solution for this at the moment.

Reply:
I was hoping that it only slowed down slice selection after the spectral axis had been converted, but it does seem a little bit slower right from the start as well. Probably not a deal-breaker but a little frustrating.